eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/PaymentOnion.scala

```diff
@@ -181,6 +181,9 @@ object OnionPaymentPayloadTlv {
 
   /** Pre-image included by the sender of a payment in case of a donation */
   case class KeySend(paymentPreimage: ByteVector32) extends OnionPaymentPayloadTlv
+
+  /** Invoice feature bits. Only included for intermediate trampoline nodes that should wait before forwarding this payment */
+  case class AsyncPaymentFeatures(features: ByteVector) extends OnionPaymentPayloadTlv
```

Comment:
We don't need anything in here, the features are unused and will take space in the onion for no good reason. I don't think there's anything wrong with an "empty" tlv, it just acts as some kind of flag. We should probably just keep this empty until we figure out if there is useful data that we should add in there:

```suggestion
  case object AsyncPayment extends OnionPaymentPayloadTlv
```

Reply:
The codec system seems to expect a case class here, it seems to work with the change here: [dc69ae5](https://github.com/ACINQ/eclair/pull/2435/commits/dc69ae5e372aba64a2795b7e25a20b620e82e60d)

I would have done this before,  but didn't find any examples of empty tlvs. I think the `provide` codec function solved it for me though.